### ISSUE TITLE
Grad CAM-Custom Architecture 

### ISSUE THREAD

**Issue by abhigoku10** [Fragjacker/Pointcloud-grad-CAM#3]
@Fragjacker @GeeeG  THanks for opensourcing the source code , i have few queries mentioned below 
1. do we have any similar implementation using pytorch framework 
2. what changes have to be made if we change architecture from Pointnet to SECOND 
3. How to extract the gradients from any layers so as to compute 

THanks in adavance

**Comment by Fragjacker** [comment]
Hello @abhigoku10 let's get to answer your questions:

1. No there's currently only this implementation available for Tensorflow.
2. The overall concept of the approach should be applicable to any kind of network that uses point clouds for input data. Which means it should not be necessary to do any changes. However you _may_ need to do some adjustments to match your network. Since this project here is part of my master thesis paper, make sure you take a look at it for more details. (Check link below)
3. In our approach we extracted the layer at the very end of the of network before it was about to be fed into a pooling layer which would cause a loss of potentially valuable information. (For details, again check the paper for details)

[Master_Thesis_paper___p-grad-CAM.pdf](https://github.com/Fragjacker/Pointcloud-grad-CAM/files/5371140/Master_Thesis_paper___p-grad-CAM.pdf)

I hope that was helpful to you.

**Comment by abhigoku10** [comment]
@Fragjacker  thanks for your quick response 

2. My custom architecture is 3D object detection so i was trying to get the gradients work for it , i am getting "NOne" value when i do backprop.
3. shall go thru the thesis in more detail

THanks for sharing
